.github/workflows/macos_debug.yml

```diff
@@ -66,14 +65,8 @@ jobs:
             -j3 \
             --output-on-failure \
             --timeout 300 \
-            --exclude-regex \
-          "tests.unit.modules.algorithms.default_construct|\
-          tests.unit.modules.algorithms.destroy|\
-          tests.unit.modules.algorithms.foreach_executors|\
-          tests.unit.modules.algorithms.max_element|\
-          tests.unit.modules.algorithms.replace_copy_if|\
-          tests.unit.modules.execution.standalone_thread_pool_executor|\
-          tests.unit.modules.resource_partitioner.used_pus"
+            --tests-regex shared_mutex2 \
+            --repeat-until-fail 10000
```

Comment:
This must be reverted before merging. I only added it to stress the test a bit more (since it typically fails on the macOS workflow).

Reply:
The build timed out because I made it do slightly too many iterations, but the test did not fail a single time: https://github.com/msimberg/pika/runs/8038639146?check_suite_focus=true.

Reply:
I've removed this temporary change.